lib/bolt/pal.rb

```diff
@@ -554,14 +549,39 @@ def get_plan_info(plan_name, with_mtime: false)
           'name'        => plan_name,
           'description' => plan.description,
           'parameters'  => parameters,
-          'module'      => mod
+          'module'      => mod,
+          'private'     => !!plan.private
         }
-        yaml_info.merge!({ 'private' => plan.private }) unless plan.private.nil?
+
         yaml_info.merge!(get_plan_mtime(yaml_path)) if with_mtime
         yaml_info
       end
     end
 
+    # Returns true if the plan is private, false otherwise.
+    #
+    # @param plan [PuppetStrings::Yard::CodeObjects::Plan] The puppet-strings plan documentation.
+    # @return [Boolean]
+    #
+    private def private_plan?(plan)
+      if plan.tag(:private)
+        value     = plan.tag(:private).text
+        api_value = value.downcase == 'true' ? 'private' : 'public'
+
+        Bolt::Logger.deprecate(
```

Comment:
```suggestion
        Bolt::Logger.deprecate_once(
```

Reply:
This would mean only the first plan to have a `@private` tag would get a warning. I think having a separate warning for each plan is helpful so you know which ones to make changes to.

Reply:
Also realizing that the plan cache means you won't see these warnings unless the plans are modified...

Reply:
That's because of a bug in `file_modified?` that I just pushed a fix for. The mtime from the plan data can be either a string if loaded from the cache or a time object if loaded from `get_plan_info`. When the file's current mtime was compared to the cache mtime, the comparison would be falsey if the mtime came from `get_plan_info` since the class is different.